packages/extension-chakra-storefront/src/hooks/cart/use-cart-shipping.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {useEffect} from 'react'
+import {
+    useShopperBasketsMutation,
+    useShippingMethodsForShipment
+} from '@salesforce/commerce-sdk-react'
+
+/**
+ * Custom hook to handle shipping method management for cart
+ * @param {Object} basket - The current basket data
+ * @returns {Object} Object containing shipping method mutations
+ */
+export const useCartShipping = (basket) => {
```

Comment:
it seems this hook is trying to set a default shipping method when there is none, can we be a little more prescriptive may be call it `useCartDefaultShipping` or something?